From b5ca749ce3a35f722812bf5d6e4ac0b60d953a49 Mon Sep 17 00:00:00 2001
From: Ayyappa Chandolu <Ayyappa.Chandolu@amd.com>
Date: Fri, 2 Mar 2018 11:25:17 +0530
Subject: [PATCH 1116/4131] pinctrl: eMMC and PinCtrl is sharing the interrupt
 no 7 for Dibber. So PinCtrl must register the interrupt hadler with SHARED
 flags.

Signed-off-by: Ayyappa Chandolu <Ayyappa.Chandolu@amd.com>
Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 drivers/acpi/resource.c       | 5 -----
 drivers/pinctrl/pinctrl-amd.c | 2 +-
 2 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/acpi/resource.c b/drivers/acpi/resource.c
index e82b5a7..d85e010 100755
--- a/drivers/acpi/resource.c
+++ b/drivers/acpi/resource.c
@@ -425,11 +425,6 @@ static void acpi_dev_get_irqresource(struct resource *res, u32 gsi,
 			triggering = trig;
 			polarity = pol;
 		}
-		if (gsi == 5) {
-			polarity = ACPI_ACTIVE_LOW;
-			pr_warning("ACPI: IRQ %d do not override to %s, %s\n", gsi,
-				   t ? "level" : "edge", p ? "low" : "high");
-		}
 	}
 
 	res->flags = acpi_dev_irq_flags(triggering, polarity, shareable);
diff --git a/drivers/pinctrl/pinctrl-amd.c b/drivers/pinctrl/pinctrl-amd.c
index b78f42a..f0e1f7c 100644
--- a/drivers/pinctrl/pinctrl-amd.c
+++ b/drivers/pinctrl/pinctrl-amd.c
@@ -896,7 +896,7 @@ static int amd_gpio_probe(struct platform_device *pdev)
 		goto out2;
 	}
 
-	ret = devm_request_irq(&pdev->dev, irq_base, amd_gpio_irq_handler, 0,
+	ret = devm_request_irq(&pdev->dev, irq_base, amd_gpio_irq_handler, IRQF_SHARED,
 			       KBUILD_MODNAME, gpio_dev);
 	if (ret)
 		goto out2;
-- 
2.7.4

